python/kvikio/kvikio/_lib/CMakeLists.txt

```diff
@@ -30,15 +30,5 @@ endif()
 rapids_cython_create_modules(
   CXX
   SOURCE_FILES "${cython_modules}"
-  LINKED_LIBRARIES kvikio::kvikio nvcomp::nvcomp
```

Comment:
This line is the main change needed to fix the current build failures in #800 (like https://github.com/rapidsai/kvikio/actions/runs/17052265746/job/48342529698?pr=800). The rest of the PR is just good cleanup.